website/addons/wiki/static/wikiPage.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+var ko = require('knockout');
+var $ = require('jquery');
+var $osf = require('osfHelpers');
+
+var mathrender = require('mathrender');
+var md = require('markdown').full;
+var mdQuick = require('markdown').quick;
+var diffTool = require('wikiDiff');
+
+var THROTTLE = 500;
+
+//<div id="preview" data-bind="mathjaxify">
+ko.bindingHandlers.mathjaxify = {
+    update: function(element, valueAccessor, allBindingsAccessor, data, context) {
+        var vm = context.$data;
+        //Need to unwrap the data in order for KO to know it's changed.
+        ko.unwrap(valueAccessor());
+
+        if(vm.allowMathjaxification() && vm.allowFullRender()) {
+            mathrender.mathjaxify('#' + element.id);
+        }
+    }
+};
+
+
+function ViewWidget(visible, version, viewText, rendered, contentURL, allowMathjaxification, allowFullRender, editor) {
+    var self = this;
+    self.version = version;
+    self.viewText = viewText; // comes from EditWidget.viewText
+    self.rendered = rendered;
+    self.visible = visible;
+    self.allowMathjaxification = allowMathjaxification;
+    self.editor = editor;
+    self.allowFullRender = allowFullRender;
+    self.renderTimeout = null;
+    self.displaySource = ko.observable('');
+
+    self.renderMarkdown = function(rawContent){
+        if(self.allowFullRender()) {
+            return md.render(rawContent);
+        } else {
+            return mdQuick.render(rawContent);
+        }
+    };
+
+    if (typeof self.editor !== 'undefined') {
+        self.editor.on('change', function () {
+            // Quick render
+            self.allowFullRender(false);
+            // Full render
+            clearTimeout(self.renderTimeout);
+
+            self.renderTimeout = setTimeout(function () {
+                self.allowFullRender(true);
+            }, THROTTLE);
+        });
+    } else {
+        self.allowFullRender(true);
+    }
+
+    self.displayText =  ko.computed(function() {
+        self.allowFullRender();
+        var requestURL;
+        if (typeof self.version() !== 'undefined') {
+            if (self.version() === 'preview') {
+                self.rendered(self.renderMarkdown(self.viewText()));
+                self.displaySource(self.viewText());
+            } else {
+                if (self.version() === 'current') {
+                    requestURL = contentURL;
+                } else {
+                    requestURL= contentURL + self.version();
+                }
+                var request = $.ajax({
```

Comment:
If you're just asking about the ajax call, then it won't happen if it's not visible because there'll be no way to change the version. It would still need to load the initial version if there were a way to, say, instantiate the page with a compare column but not a view column (which I don't think is currently possible, but might be before the end of the week, depending). So I think the ajax request should happen regardless. 

If we want rendering not to happen when it's not visible (which is sane), we'd probably want to add the visibility check to the rendering function.